lib/http/HttpRequestApple.mm

```diff
@@ -0,0 +1,153 @@
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "mat/config.h"
+
+// Assume that if we are compiling with MSVC, then we prefer to use Windows HTTP stack,
+// e.g. WinInet.dll or Win 10 HTTP client instead
+#if defined(MATSDK_PAL_CPP11) && !defined(_MSC_VER) && defined(HAVE_MAT_DEFAULT_HTTP_CLIENT)
+
+#import <Foundation/Foundation.h>
+#import <CFNetwork/CFNetwork.h>
+
+#include "HttpClient_Apple.hpp"
+#include "HttpRequestApple.hpp"
+#include "utils/StringUtils.hpp"
+
+namespace MAT_NS_BEGIN {
+
+static dispatch_once_t once;
```

Comment:
These are still global statics, but I understand there are challenges in making it member-variable for this class - https://stackoverflow.com/questions/13856037/can-i-declare-dispatch-once-t-predicate-as-a-member-variable-instead-of-static/13858628#13858628. Also, there are issues in making it static member of the class. Will approve it if it is not practically possible to do either.
Also I understand end to end tests have been done - events are flowing, and there is no memory leak observed in this part of code.